### PR TITLE
Added automated workflow for pushing codeflare-operator image

### PR DESCRIPTION
**Pull request by Bobbins228**
closes: [#13](https://github.com/red-hat-data-services/distributed-workloads/issues/13)
## Description

Created a Github actions workflow for automatically pushing the `opendatahub/codeflare-operator` image and updating the compatibility matrix in the README and component versions in the MakeFile
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/build-and-push.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        sed -i -E "s/(.*CODEFLARE_SDK_VERSION \?= ).*/\1$CODEFLARE_SDK_VERSION/" Makefile
+        sed -i -E "s/(.*INSTASCALE_VERSION \?= ).*/\1${{ env.INSTASCALE_RELEASE_TAG }}/" Makefile
+    
+    - name: Image Build and Push
+      run: |
+        make build
+        make image-build -e IMG=quay.io/opendatahub/codeflare-operator:${{ env.RELEASE_TAG }}
+        make image-push -e IMG=quay.io/opendatahub/codeflare-operator:${{ env.RELEASE_TAG }}
+
+    - name: Cleanup the go.mod and go.sum
```
Comment (anishasthana) [inline .github/workflows/build-and-push.yaml:68]: Why are we running this after the image build?

File: .github/workflows/build-and-push.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    - name: Wait until PR with code changes is merged
+      run: |
+        timeout 3600 bash -c 'until [[ $(gh pr view '${{ env.PR_BRANCH_NAME }}' --json state --jq .state) == "MERGED" ]]; do sleep 5 && echo "$(gh pr view '${{ env.PR_BRANCH_NAME }}' --json state --jq .state)"; done'
+      env:
+        GITHUB_TOKEN: ${{ github.TOKEN }}
+
+    - name: Delete remote branch
+      run: |
+        git push origin --delete ${{ env.PR_BRANCH_NAME }}
```
Comment (anishasthana) [inline .github/workflows/build-and-push.yaml:95]: Missing new line

File: .github/workflows/build-and-push.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# This workflow will build the Codeflare Operator image and push it to the opendatahub image registry as well as update versions in the MakeFile and README
```
Comment (anishasthana) [inline .github/workflows/build-and-push.yaml:1]:
```suggestion
# This workflow will build the CodeFlare Operator image and push it to the opendatahub image registry as well as update versions in the MakeFile and README
```

File: .github/workflows/build-and-push.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        password: ${{ secrets.QUAY_BOT_PASSWORD }}
+        registry: quay.io
+
+    - name: Get Upstream Release Tags 
+      id: release-tags
+      run: |
+        release_tag=$(gh release view -R github.com/project-codeflare/codeflare-operator --json tagName | jq -r '.tagName')
+        echo "RELEASE_TAG=$release_tag" >> $GITHUB_ENV
+      env:
+        GITHUB_TOKEN: ${{ github.TOKEN }}
```
Comment (VanillaSpoon) [inline .github/workflows/build-and-push.yaml:37]: I can't see anywhere in the [DOCS](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) about whether `github.token` is case sensitive. But if a run-through was successful then lgtm :)

Comment (Bobbins228) [reply .github/workflows/build-and-push.yaml]: AFAIK for `github.token` it is not case sensitive but I've updated it anyway to be in line with how its used in the docs :) Thanks!

**Review by anishasthana — APPROVED** [review]

**Pull request merged**